server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/ImportTable.java

```diff
@@ -127,7 +139,55 @@ public void checkVersions(Master env) throws AcceptableThriftTableOperationExcep
 
   @Override
   public void undo(long tid, Master env) throws Exception {
-    Utils.unreserveHdfsDirectory(env, new Path(tableInfo.exportDir).toString(), tid);
+    for (ImportedTableInfo.DirectoryMapping dm : tableInfo.directories) {
+      Utils.unreserveHdfsDirectory(env, new Path(dm.exportDir).toString(), tid);
+    }
+
     Utils.unreserveNamespace(env, tableInfo.namespaceId, tid, false);
   }
+
+  static List<ImportedTableInfo.DirectoryMapping> parseExportDir(String exportDir) {
+    if (exportDir == null || exportDir.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    String[] exportDirs = exportDir.split(",");
+    List<ImportedTableInfo.DirectoryMapping> dirs = new ArrayList<>(exportDirs.length);
+    for (String ed : exportDirs) {
+      log.info("Extracted import directory: {}", ed);
+      ImportedTableInfo.DirectoryMapping dir = new ImportedTableInfo.DirectoryMapping();
+      dir.exportDir = ed;
+      dirs.add(dir);
+    }
+    return dirs;
+  }
+
+  Path findExportFile(Master env) throws AcceptableThriftTableOperationException {
```

Comment:
This code looks very familiar to something I saw in TableOperationsImpl.  Can we somehow merge or does this have to remain separate.

Reply:
I removed this method and modified the method call to be able to reuse the implementation from `TableOperationsImpl`. There was a bit of type contortion that I had to do, but I think it works.